README.md

```diff
@@ -25,7 +25,7 @@ steps:
 - run: npm test
 ```
 
-By default, the branch or tag ref that triggered the workflow will be checked out, `${{ github.token }}` will be used for any Git server authentication. If you wish to check out a different branch, a different repository or use different token to checkout, specify that using `with.ref`, `with.repository` and `with.token`:
+By default, the sha for the given branch or tag ref that triggered the workflow will be checked out, (leaving the repository in detached HEAD state). `${{ github.token }}` will be used for any Git server authentication. If you wish to check out a different branch, a different repository or use a different token to checkout, specify that using `with.ref`, `with.repository` or `with.token` respectively:
```

Comment:
working on a v2 version of checkout with some behavioral changes. it will not checkout detached head anymore for branches

i'll update the docs with the behavior, in that [pr](https://github.com/actions/checkout/pull/70)

Reply:
fantastic!!

Reply:
Just wanted to feedback that v2-beta checks out the branch and works fine for me. Thanks!